src/SMU-Keysight_E3631A/main.py

```diff
@@ -159,35 +165,42 @@ def poweroff(self):
         self.port.write("OUTP:STAT OFF")
 
     def apply(self):
```

Comment:
instead of checking `float(self.value)` in every `if` clause, you could write `self.value = float(self.value)` in the first line of `apply`